src/long.ts

```diff
@@ -176,6 +176,17 @@ export class Long extends BSONValue {
   /** Minimum signed value. */
```

Comment:
contstuctor overloads and Object.assign improvement

Reply:
We should remove the `!` from the low,high,unsigned properties, Typescript will help assert that if there's any new properties in the Long class that they must be set in the CTOR

Reply:
Yep, good call!

Reply:
There's still `!` on the properties because we still use the assign for fromString.

Something to consider, using a ternary that results in "always" fixed types might work well here, then you can just assign the results to the properties unconditionally

```js
    const {
      low: lowNumber,
      high: highNumber,
      unsigned: unsignedBoolean
    } = typeof low === 'string'
      ? Long.fromString(low, Boolean(high))
      : typeof low === 'bigint'
      ? Long.fromBigInt(low, Boolean(high))
      : typeof high === 'boolean'
      ? { low, high: 0, unsigned: high }
      : { low, high: high ?? 0, unsigned: Boolean(unsigned) };
```

Reply:
I see, didn't realize we were going to make any changes in the `string` branch